FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -2196,6 +2203,15 @@ - (BOOL)useUserAccessGroup:(NSString *_Nullable)accessGroup
   return YES;
 }
 
+- (BOOL)useUserAccessGroup:(NSString *_Nullable)accessGroup
+                     error:(NSError *_Nullable *_Nullable)outError {
+  // Make sure that the keychain has been initialized.
+  dispatch_semaphore_wait(_initializingKeychain, DISPATCH_TIME_FOREVER);
```

Comment:
This approach of splitting internal and external usage would work just as well with your `dispatch_sync` approach and would avoid needing to keep an extra semaphore (or keep waiting/re-signaling it).